server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -776,6 +789,35 @@ public ClusterState execute(BatchExecutionContext<StartedShardUpdateTask> batchE
             return maybeUpdatedState;
         }
 
+        private ClusterState maybeRemoveIndexRefreshBlocks(
+            ClusterState clusterState,
```

Comment:
should we document that the cluster state here is already updated with the `STARTED` state of search shards?

Reply:
Done in https://github.com/elastic/elasticsearch/pull/120807/commits/b766d8e7f603de554ab684c3123a2a2a26196562